sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -1664,3 +1664,21 @@ case class Call(
   override protected def withNewChildInternal(newChild: LogicalPlan): Call =
     copy(procedure = newChild)
 }
+
+/**
+ * The logical plan of the SHOW PROCEDURES command.
+ */
+case class ShowProcedures(
```

Comment:
I think a better way to implement commands today is to follow `DescribeRelationJsonCommand`:
1. extends `UnaryRunnableCommand` so that it merges logical and physical plans, while still leverages the v2 framework to resolve table/namespace
2. put in `sql/core` as `UnaryRunnableCommand` is in `sql/core`. The parser change should be put in `SparkSqlAstBuilder`